src/functions/wiki-documentation/upload-file/handler.ts

```diff
@@ -0,0 +1,42 @@
+import { middyfy } from "@libs/lambda";
+import { APIGatewayProxyEvent, APIGatewayProxyHandler } from "aws-lambda";
+import { generatePreSignedUrl } from "src/services/s3-file-service";
+
+/**
+ * Handler for creating a presigned url for the following file upload.
+ *
+ * @param event - API Gateway event.
+ * @returns Response object with status code.
+ */
+const uploadFileHandler: APIGatewayProxyHandler = async (event: APIGatewayProxyEvent) => {
+  const { body } = event;
+  const { path, contentType } = (typeof body === "string" ? JSON.parse(body) : body);
+
+  if (!path) {
+    return {
+      statusCode: 400,
+      body: JSON.stringify({
+        code: 400,
+        message: "Invalid request body.",
+      })
+    };
+  }
+
+  try {
+    const presignedUrl = await generatePreSignedUrl(path, "put", contentType);
```

Comment:
Why the presigned request is created in the backend? Usually these are done in the UI and we are using seperate lambda for uploading the actual files